### PR TITLE
Chapter 9: Advanced Login

### PR DESCRIPTION
**Pull request by megatimranikram**

### CODE REVIEW

**Comment by megatimranikram** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -38,4 +37,4 @@ gem "tzinfo-data", platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 group :development, :test do
   gem "rubocop", "~> 0.74.0", require: false
   gem "rubocop-rails", "~> 2.3.2", require: false
-end
+end
```
Comment (saobv-2223) [inline Gemfile]:
```suggestion
end

```

File: .gitignore [hunk 2]
```diff
@@ -46,4 +46,4 @@ yarn-debug.log*
 
 config/settings.local.yml
 config/settings/*.local.yml
-config/environments/*.local.yml
+config/environments/*.local.yml
```
Comment (saobv-2223) [inline .gitignore]:
```suggestion
config/environments/*.local.yml

```

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -3,4 +3,4 @@ def full_title page_title
     base_title = "Ruby on Rails Tutorial Sample App"
     page_title.present? ? [page_title, base_title].join(" | ") : base_title
   end
-end
+end
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]:
```suggestion
end

```

File: config/application.rb [hunk 4]
```diff
@@ -10,4 +10,4 @@ class Application < Rails::Application
     config.i18n.available_locales = [:en, :ru]
     config.i18n.default_locale = :en
   end
-end
+end
```
Comment (saobv-2223) [inline config/application.rb]:
```suggestion
end

```

File: config/initializers/config.rb [hunk 5]
```diff
@@ -52,4 +52,4 @@
     #   required(:email).filled(format?: EMAIL_REGEX)
     # end
   
-  end
+  end
```
Comment (saobv-2223) [inline config/initializers/config.rb]:
```suggestion
  end
  
```

File: config/settings.yml [hunk 6]
```diff
@@ -12,3 +12,5 @@ validations:
     regex: !ruby/regexp '/\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i'
   password:
     min_length: 8
+
+remember_me: "1"
```
Comment (saobv-2223) [inline config/settings.yml]:
```suggestion
remember_me: "1"

```

File: db/schema.rb [hunk 7]
```diff
@@ -10,14 +10,14 @@
...
   create_table "users", options: "ENGINE=InnoDB DEFAULT CHARSET=latin1", force: :cascade do |t|
     t.string "name"
     t.string "email"
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
     t.string "password_digest"
-    t.index ["email"], name: "index_users_on_email", unique: true
+    t.string "remember_digest"
   end
 end
```
Comment (saobv-2223) [inline db/schema.rb]:
```suggestion
end

```

**Comment by megatimranikram** [comment]
ready

File: Gemfile (hunk 1 above)
Comment (megatimranikram) [reply Gemfile]: done sir

File: .gitignore (hunk 2 above)
Comment (megatimranikram) [reply .gitignore]: done sir

File: app/helpers/application_helper.rb (hunk 3 above)
Comment (megatimranikram) [reply app/helpers/application_helper.rb]: done sir

File: config/application.rb (hunk 4 above)
Comment (megatimranikram) [reply config/application.rb]: done sir

File: config/initializers/config.rb (hunk 5 above)
Comment (megatimranikram) [reply config/initializers/config.rb]: done sir

File: config/settings.yml (hunk 6 above)
Comment (megatimranikram) [reply config/settings.yml]: done sir

File: db/schema.rb (hunk 7 above)
Comment (megatimranikram) [reply db/schema.rb]: done sir

**Comment by manhnt-0960** [comment]
merged

**Pull request merged**